router/batchrouter/batchrouter.go

```diff
@@ -455,7 +455,7 @@ func (brt *HandleT) recordDeliveryStatus(batchDestination DestinationT, err erro
 }
 
 func (brt *HandleT) initWorkers() {
-	for i := 0; i < noOfWorkers; i++ {
+	for i := 0; i < brt.noOfWorkers; i++ {
```

Comment:
One more thing, not related to this PR. 

initWorkers need not be called inside a goroutine. Not a problem, but unncessary.

Can you check with @psrikanth88 ?